delfin/drivers/hitachi/vsp/vsp_stor.py

```diff
@@ -0,0 +1,257 @@
+        try:
+            pools_info = self.rest_handler.get_all_pools()
+            pool_list = []
+            pools = pools_info.get('data')
+            for pool in pools:
+                status = self.POOL_STATUS_MAP.get(
+                    pool.get('poolStatus'),
+                    constants.StoragePoolStatus.ABNORMAL
+                )
+                storage_type = constants.StorageType.BLOCK
+                total_cap = \
+                    int(pool.get('totalPoolCapacity')) * units.Mi
+                free_cap = int(
+                    pool.get('availableVolumeCapacity')) * units.Mi
+                used_cap = total_cap - free_cap
+                p = {
+                    'name': pool.get('poolName'),
+                    'storage_id': self.storage_id,
+                    'native_storage_pool_id': str(pool.get('poolId')),
+                    'description': 'Hitachi VSP Pool',
+                    'status': status,
+                    'storage_type': storage_type,
+                    'subscribed_capacity': int(total_cap),
+                    'total_capacity': int(total_cap),
+                    'used_capacity': int(used_cap),
+                    'free_capacity': int(free_cap),
+                }
+                pool_list.append(p)
+
+            return pool_list
+        except exception.DelfinException as err:
+            err_msg = "Failed to get pool metrics from hitachi vsp: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise err
+        except Exception as e:
+            err_msg = "Failed to get pool metrics from hitachi vsp: %s" % \
+                      (six.text_type(e))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_volumes(self, context):
+        try:
+            wwn = ''
+            volumes_info = self.rest_handler.get_all_volumes()
+
+            volume_list = []
+            volumes = volumes_info.get('data')
+            for volume in volumes:
+                if volume.get('emulationType') == 'NOT DEFINED':
+                    continue
+                orig_pool_id = volume.get('poolId')
+                compressed = False
+                deduplicated = False
+                if volume.get('dataReductionMode') == \
+                        'compression_deduplication':
+                    deduplicated = True
+                    compressed = True
+                if volume.get('dataReductionMode') == 'compression':
+                    compressed = True
+                if volume.get('status') == 'NML':
+                    status = 'normal'
+                else:
+                    status = 'abnormal'
+
+                vol_type = constants.VolumeType.THICK
+                for voltype in volume.get('attributes'):
+                    if voltype == 'HTI':
+                        vol_type = constants.VolumeType.THIN
+
+                total_cap = \
```

Comment:
`total_cap` and `used_cap` seems to be same.

Reply:
there is no used_cap or free_cap,so i set them one value

Reply:
But now total_cap and used_cap are set one value